packages/shared/components/popups/MintNftFormPopup.svelte

```diff
@@ -0,0 +1,158 @@
+        const valid = validate()
+        if (valid) {
+            setMintNftDetails({
+                id: undefined,
+                standard: TokenStandard.IRC27,
+                version: undefined,
+                type,
+                uri,
+                name,
+                collectionId,
+                collectionName,
+                royalties,
+                issuerName,
+                description,
+                attributes,
+            })
+            openPopup({
+                type: 'mintNftConfirmation',
+                overflow: true,
+            })
+        }
+    }
+
+    function handleSelectNftType(item: { label: MimeType; value: MimeType }) {
+        type = item.value
+    }
+
+    function validate(): boolean {
+        // TODO: implement validation
+        return true
+    }
+
+    onMount(async () => {
+        try {
+            await _onMount()
+        } catch (err) {
+            handleError(error)
+        }
+    })
+</script>
+
+<div class="space-y-6">
+    <Text type="h4" fontSize="18" lineHeight="6" fontWeight={FontWeight.semibold}>
+        {localize('popups.mintNftForm.title')}
+    </Text>
+
+    <div class="space-y-4 max-h-100 scrollable-y overflow-x-hidden flex-1">
+        <Dropdown
+            bind:value={type}
+            onSelect={handleSelectNftType}
+            label={localize('popups.mintNftForm.inputs.type')}
+            placeholder={localize('popups.mintNftForm.inputs.type')}
+            items={nftTypeOptions}
+            fontSize="sm"
+            lineHeight="140"
+            fontWeight={FontWeight.medium}
+            error={typeError}
+        />
+        <TextInput
+            bind:value={uri}
+            label={localize('popups.mintNftForm.inputs.uri')}
+            placeholder={localize('popups.mintNftForm.inputs.uri')}
+            error={uriError}
+        />
+        <TextInput
+            bind:value={name}
+            label={localize('popups.mintNftForm.inputs.name')}
+            placeholder={localize('popups.mintNftForm.inputs.name')}
+            error={nameError}
+        />
+        <optional-inputs class="flex flex-row flex-wrap gap-4">
+            <OptionalInput
+                bind:value={collectionId}
+                label={localize('popups.mintNftForm.inputs.collectionId')}
+                description={localize('tooltips.mintNftForm.collectionId')}
+                fontSize="14"
+            />
+            <OptionalInput
+                bind:value={collectionName}
+                label={localize('popups.mintNftForm.inputs.collectionName')}
+                description={localize('tooltips.mintNftForm.collectionName')}
+                fontSize="14"
+            />
+            <OptionalInput
+                bind:value={royalties}
+                label={localize('popups.mintNftForm.inputs.royalties')}
+                description={localize('tooltips.mintNftForm.royalties')}
+                fontSize="14"
+            />
```

Comment:
For this we need another type of input because it is a list of objects. 
I'm gonna implement a component for this

Reply:
Sounds good; I'd like to hear more about your plan for this. I was just implementing all of the NFT fields as OptionalInputs as a first pass. But as I was doing this I wondered about both royalties and attributes and what the best approach for them would be.